libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/MapboxCarContext.kt

```diff
@@ -1,37 +1,101 @@
 package com.mapbox.androidauto.car
 
 import androidx.car.app.CarContext
-import androidx.lifecycle.LifecycleOwner
-import com.mapbox.androidauto.car.feedback.core.CarFeedbackOptions
-import com.mapbox.androidauto.car.feedback.core.CarFeedbackPollProvider
-import com.mapbox.androidauto.car.navigation.speedlimit.SpeedLimitOptions
-import com.mapbox.androidauto.car.preview.CarRouteOptionsInterceptor
+import androidx.car.app.navigation.NavigationManager
+import androidx.lifecycle.Lifecycle
+import com.mapbox.androidauto.MapboxCarNavigationManager
 import com.mapbox.androidauto.car.preview.CarRoutePreviewRequest
-import com.mapbox.androidauto.car.search.CarPlaceSearchOptions
-import com.mapbox.androidauto.car.settings.CarSettingsStorage
+import com.mapbox.androidauto.car.settings.MapboxCarStorage
 import com.mapbox.androidauto.deeplink.GeoDeeplinkPlacesListOnMapProvider
+import com.mapbox.androidauto.internal.car.context.MapboxCarContextOwner
+import com.mapbox.androidauto.internal.car.context.mapboxCarService
+import com.mapbox.androidauto.notification.CarNotificationInterceptor
 import com.mapbox.androidauto.screenmanager.MapboxScreenManager
 import com.mapbox.maps.MapboxExperimental
 import com.mapbox.maps.extension.androidauto.MapboxCarMap
-import kotlinx.coroutines.flow.MutableStateFlow
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 
+/**
+ * This is the entry point for Mapbox Navigation in Android Auto. Create this object and then you
+ * have access to services and options for customization.
+ *
+ * @param lifecycle used to access the [CarContext].
+ * @param mapboxCarMap controls the Mapbox car map surface.
+ */
 @OptIn(MapboxExperimental::class)
 class MapboxCarContext(
-    val carContext: CarContext,
-    val lifecycleOwner: LifecycleOwner,
+    /**
+     * Gives access to the [Lifecycle].
+     */
+    val lifecycle: Lifecycle,
+    /**
+     * [MapboxCarMap] controls the Mapbox car map surface
+     */
     val mapboxCarMap: MapboxCarMap,
-    val feedbackPollProvider: CarFeedbackPollProvider = CarFeedbackPollProvider(),
-    val carFeedbackOptions: CarFeedbackOptions = CarFeedbackOptions.Builder().build(),
-    routeOptionsInterceptor: CarRouteOptionsInterceptor = CarRouteOptionsInterceptor { it },
-    val carPlaceSearchOptions: CarPlaceSearchOptions = CarPlaceSearchOptions.Builder().build(),
 ) {
-    val carSettingsStorage = CarSettingsStorage(carContext)
-    val mapboxScreenManager = MapboxScreenManager(carContext, lifecycleOwner)
+    private val carContextOwner = MapboxCarContextOwner(lifecycle)
```

Comment:
`CarContext` can also be retrieved from `MapboxCarMap`, might be easier.

Reply:
[`MapboxCarMap`](https://github.com/mapbox/mapbox-maps-android/blob/3beb4c8b63056508656f97abaa56b3bca962951e/extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt#L71-L77) requires a setup call.

Still need a trigger for when the CarContext has been created

Reply:
`MapboxCarContextOwner` could potentially move up to the map sdk. But I've made it internal because I'm not entirely happy with it yet..

Reply:
Resolving this one because `MapboxCarMap` may change how it manages the `CarContext`.

`MapboxCarMap` does not convert it `carContext` to `null` after it is destroyed. `MapboxCarContextOwner` is still internal and maybe we move that implementation to the maps sdk.